### PR TITLE
utils_v2v:update check_log function

### PR DESCRIPTION
**Pull request by chloerh**
get parameters (log pattern to search for, log is expected or not)
directly from config file.For expected message list, return False if any
message is NOT shown.For not expected message list, return False if any
message is shown.

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

File: virttest/utils_v2v.py [hunk 1]
```diff
@@ -742,31 +742,45 @@ def import_vm_to_ovirt(params, address_cache, timeout=600):
     return True
 
 
-def check_log(log, content_list, expect=True):
+def check_log(params, log):
     """
-    Check if Error Message exists in v2v log
+    Check if error/warning meets expectation in v2v log
```
Comment (xutian) [inline virttest/utils_v2v.py]: what's  params ? please add doc string here .

Comment (xutian) [inline virttest/utils_v2v.py]: I think that better way is o give a pattern list  for your message,  if message match pattern return True, else return False.

File: virttest/utils_v2v.py [hunk 2]
```diff
@@ -742,31 +742,45 @@ def import_vm_to_ovirt(params, address_cache, timeout=600):
...
-            found = True
-            logging.info('Found log: \n%s' % search.group(0))
-    # Expect to find log but failed
-    if not found and expect:
-        logging.error('Log not found: %s' % '\n'.join(content_list))
-        return False
-    return True
+
+    def _check_log(content_list, expect=True):
+        for content in content_list:
```
Comment (xutian) [inline virttest/utils_v2v.py]: Please don't split content and match it line by line, please use re.M to match multi-lines.

File: virttest/utils_v2v.py [hunk 3]
```diff
@@ -742,31 +742,45 @@ def import_vm_to_ovirt(params, address_cache, timeout=600):
...
+            line = '\s*'.join(content.split())
+            expected = 'expected' if expect else 'not expected'
+            logging.info('Searching for %s log: %s' % (expected, content))
+            pattern = re.compile(line)
+            search = re.search(pattern, log)
+            if search:
+                logging.info('Found log: %s', search.group(0))
+                # Found log not expect
```
Comment (xutian) [inline virttest/utils_v2v.py]: useless comment

File: virttest/utils_v2v.py [hunk 4]
```diff
@@ -742,31 +742,45 @@ def import_vm_to_ovirt(params, address_cache, timeout=600):
...
+                if not expect:
+                    return False
+            else:
+                logging.info('Not find log: %s', content)
+                # Not find expected log
```
Comment (xutian) [inline virttest/utils_v2v.py]: same as above.

**Comment by chloerh** [comment]
Recent test result (with this patch)

esx.vm.5_1.linux.7_4.x86_64.raw_f.NFS | 33 分 | Passed
esx.vm.5_1.windows.win10.x86_64.raw_f.NFS | 26 分 | Passed
esx.vm.5_1.windows.win2008r2.x86_64.raw_f.NFS | 30 分 | Passed
esx.vm.5_1.windows.win2012.x86_64.raw_f.NFS | 24 分 | Passed
esx.vm.5_5.linux.6_9.i386.raw_f.NFS | 32 分 | Passed
esx.vm.5_5.windows.win10.i386.raw_f.NFS | 26 分 | Passed
esx.vm.5_5.windows.win2008.x86_64.raw_f.NFS | 27 分 | Passed
esx.vm.5_5.windows.win8_1.x86_64.raw_f.NFS | 29 分 | Passed
esx.vm.6_0.linux.6_9.x86_64.raw_f.NFS | 33 分 | Passed
esx.vm.6_0.windows.win2012r2.x86_64.raw_f.NFS | 23 分 | Passed
esx.vm.6_0.windows.win7.x86_64.raw_f.NFS | 24 分 | Passed
esx.vm.6_0.windows.win8.i386.raw_f.NFS | 23 分 | Passed
esx.vm.6_5.linux.7_4.x86_64.raw_f.NFS | 16 秒 | Regression
esx.vm.6_5.windows.win10.x86_64.raw_f.NFS | 27 分 | Passed
esx.vm.6_5.windows.win2012r2.x86_64.raw_f.NFS | 25 分 | Passed
esx.vm.6_5.windows.win7.i386.raw_f.NFS | 25 分 | Passed
kvm.linux.5_11.i386.raw_f.NFS | 11 分 | Passed
kvm.linux.6_9.i386.raw_f.NFS | 14 分 | Passed
kvm.linux.6_9.x86_64.qcow2_f.NFS | 14 分 | Passed
kvm.linux.7_4.x86_64.qcow2_f.NFS | 15 分 | Passed
kvm.windows.win10.i386.qcow2_f.NFS | 16 分 | Passed
kvm.windows.win2008.i386.raw_f.NFS | 16 分 | Passed
kvm.windows.win2016.x86_64.qcow2_f.NFS | 21 分 | Fixed
kvm.windows.win7.i386.qcow2_f.NFS | 22 分 | Skipped
kvm.windows.win8.x86_64.raw_f.NFS | 18 分 | Passed
kvm.windows.win8_1.i386.qcow2_f.NFS | 18 分 | Passed
xen.vm.hvm.linux.6_9.x86_64.raw_f.NFS | 25 分 | Fixed
xen.vm.pv.linux.6_9.i386.raw_f.NFS | 24 分 | Fixed

**Comment by weikunzz** [comment]
The code works good to me

**Pull request merged**